### PR TITLE
ObservableReleasesClient.GetAll should use overloads properly

### PR DESCRIPTION
**Pull request by shiftkey**
See comment: https://github.com/octokit/octokit.net/pull/1213#issuecomment-200981143

 - [x] fix impacted tests

### CODE REVIEW

File: Octokit.Reactive/Clients/ObservableReleasesClient.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public IObservable<Release> GetAll(string owner, string name)
             Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
             Ensure.ArgumentNotNullOrEmptyString(name, "name");
```
Comment (dampir) [inline Octokit.Reactive/Clients/ObservableReleasesClient.cs]:
```cs
`  Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
              Ensure.ArgumentNotNullOrEmptyString(name, "name");`
Should we save these checks if we them in overloaded method too?

Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableReleasesClient.cs]: Personally I'd lean towards "no", but I'll let @Haacked chime in here as well and see if we agree or not on this...

Comment (Haacked) [inline Octokit.Reactive/Clients/ObservableReleasesClient.cs]: I think of these as being like contracts. I want them as close to the call-site as possible. Semantically, I think of these arguments being invalid for this method.

Eventually, we'll replace many of these with Fody NullGuard which will throw the exception at the call site.

Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableReleasesClient.cs]: > I think of these as being like contracts. I want them as close to the call-site as possible. 

:thumbsup:

**Comment by shiftkey** [comment]
Just rebased this due to a merge conflict. @ryangribble can I get a :thumbsup: here?

**Comment by ryangribble** [comment]
> Just rebased this due to a merge conflict. @ryangribble can I get a :thumbsup: here?

Given the tests are passing, I wasn't going to look too deeply, but I did notice that the title of commit and PR "ensure observable method calls overload with ApiOptions" isnt actually evident in the code changes - which are only tidying up whitespace formatting and changing the mock asertion to ANY dictionary rather than empty dictionary.

Im assuming the actual change to call the correct overload was already in master, and the rebase is hiding that here?  Should the PR be reworded to "tidy up/tweak test" ?

**Comment by shiftkey** [comment]
@ryangribble yeah, looks like #1212 also touched these files. Given the difference is now minor, I'll skip it.

**Pull request closed without merging**